DDOT_module/child_classes/correction_data_generator.py

```diff
@@ -0,0 +1,397 @@
+
+            # check whether qubits are properly sorted
+            sorted_quest = True
+            if cluster_i == cluster_j:
+                pass
+            else:
+                for ccc1 in cluster_j:
+                    for ccc0 in cluster_i:
+                        if ccc0 > ccc1:
+                            sorted_quest = False
+
+            whole_marginal = sorted(anf.lists_sum(cluster_i, cluster_j))
+            if not sorted_quest:
+                # TODO: swapping qubits, does it make sense?
+                averaged_matrix = averaged_matrix_clusters_i_j
+                import QREM.povmtools as qrem_pt
+
+                qubits_in_here = cluster_i + cluster_j
+                sorted_qubits_in_here = dict(enumerate(sorted(qubits_in_here)))
+
+                rev_map = anf.get_enumerated_rev_map_from_indices(cluster_i + cluster_j)
+
+                qubits_in_here_dict = dict(enumerate(qubits_in_here))
+
+                while qubits_in_here_dict != sorted_qubits_in_here:
+                    for index_qubit_hehe in range(len(qubits_in_here) - 1):
+                        if qubits_in_here[index_qubit_hehe] < qubits_in_here[index_qubit_hehe + 1]:
+                            pass
+                        elif qubits_in_here[index_qubit_hehe + 1] < qubits_in_here[
+                            index_qubit_hehe]:
+                            averaged_matrix = qrem_pt.permute_matrix(averaged_matrix,
+                                                                     len(whole_marginal),
+                                                                     [index_qubit_hehe + 1,
+                                                                      index_qubit_hehe + 2])
+
+                            anf.cool_print('Swapping qubits:', )
+                            print(qubits_in_here[index_qubit_hehe],
+                                  qubits_in_here[index_qubit_hehe + 1], index_qubit_hehe,
+                                  index_qubit_hehe + 1)
+
+                            qubits_in_here[index_qubit_hehe], qubits_in_here[
+                                index_qubit_hehe + 1] = qubits_in_here[index_qubit_hehe + 1], \
+                                                        qubits_in_here[index_qubit_hehe]
+
+                    qubits_in_here_dict = dict(enumerate(qubits_in_here))
+
+
+            else:
+                averaged_matrix = averaged_matrix_clusters_i_j
+
+            correction_matrix = np.linalg.inv(averaged_matrix)
+
+            string_marginal = ''.join(['q%s' % qqq for qqq in whole_marginal])
+            self._correction_indices[string_pair] = string_marginal
+            self._correction_indices['q%s' % i] = string_marginal
+            self._correction_indices['q%s' % j] = string_marginal
+            self._noise_matrices[string_marginal] = averaged_matrix
+            self._correction_matrices[string_marginal] = correction_matrix
+
+            if calculate_mitigation_errors:
```

Comment:
The variable is named `calculate_mitigation_errors` but I do not see any calculations in this block. Maybe you should move this block near the calculations?